docs/articles/documentation/using-testcafe/programming-interface/runner.md

```diff
@@ -319,7 +319,7 @@ reporter([ name | { name, output }]) → this
 
 Parameter                | Type                        | Description                                     | Default
 ------------------------ | --------------------------- | ----------------------------------------------- | --------
-`name`                   | String              | The name of the [reporter](../common-concepts/reporters.md) to use.
+`name`                   | String &#124; Object &#124; Function      | The name of the [reporter](../common-concepts/reporters.md), a custom reporter object, or a function that returns this object.
```

Comment:
As long I remember, we discussed that `name` parameter can have only `string` type.
So, the example should be rewritten as follows:
```js
import { createTestCafe } from 'testcafe';
const customReporter = () => {
    return {
        async reportTaskStart (startTime, userAgents, testCount) { /* ... */ },
        async reportFixtureStart (name, path, meta) { /* ... */ },
        async reportTestDone (name, testRunInfo, meta) { /* ... */ },
        async reportTaskDone (endTime, passed, warnings, result) { /* ... */ }
    };
};
const testcafe = await createTestCafe(/* [...] */);
const runner   = testcafe.createRunner();
await runner.reporter(customReporter);
```

@AndreyBelym take a look please

Reply:
Yep, that's how we use it in our tests:

https://github.com/DevExpress/testcafe/blob/master/test/server/runner-test.js#L950

Reply:
We can't pass just a reporter object right now.